cpp/src/arrow/memory_pool.cc

```diff
@@ -300,6 +302,8 @@ class JemallocAllocator {
       dallocx(ptr, MALLOCX_ALIGN(kAlignment));
     }
   }
+
+  static void ReleaseUnused() {}
```

Comment:
Apparently, we can ask jemalloc to explicitly purge unused pages: https://github.com/jemalloc/jemalloc/issues/1190

Reply:
I added a jemalloc implementation.  It does not appear to do anything in my manual testing but it is possible that jemalloc was simply just already releases pages to the OS.  Since this is best-effort anyways I think it is ok.